compiler-cli/src/fix.rs

```diff
@@ -1,13 +1,57 @@
+use std::sync::Arc;
+
 use camino::{Utf8Path, Utf8PathBuf};
 use gleam_core::{
+    analyse::TargetSupport,
+    build::{Codegen, Compile, Mode, Options},
     error::{FileIoAction, FileKind},
-    Error, Result,
+    type_,
+    warning::VectorWarningEmitterIO,
+    Error, Result, Warning,
 };
+use hexpm::version::Version;
+
+use crate::{build, cli};
+
+enum FixResult {
+    FixedSomething,
+    NothingToFix,
+}
 
 pub fn run() -> Result<()> {
-    for path in crate::fs::gleam_files_excluding_gitignore(Utf8Path::new(".")) {
-        fix_file(path)?;
-    }
+    // When running gleam fix we want all the compilation warnings to be hidden,
+    // at the same time we need to access those to apply the fixes: so we
+    // accumulate those into a vector.
+    let warnings = Arc::new(VectorWarningEmitterIO::new());
+    let _built = build::main_with_warnings(
+        Options {
+            root_target_support: TargetSupport::Enforced,
+            warnings_as_errors: false,
+            codegen: Codegen::DepsOnly,
+            compile: Compile::All,
+            mode: Mode::Dev,
+            target: None,
+            no_print_progress: false,
+        },
+        build::download_dependencies(cli::Reporter::new())?,
+        warnings.clone(),
+    )?;
+    let warnings = warnings.take();
+
+    let result = fix_minimum_required_version(warnings)?;
+
+    match result {
+        FixResult::FixedSomething => println!("Your Gleam code has been fixed!"),
+        FixResult::NothingToFix => println!("Nothing to fix."),
+    };
```

Comment:
Do you think we should be more precise about what has been fixed?

Reply:
Yeah good idea!